### ISSUE TITLE
error on npm install after cloning repo

### ISSUE THREAD

**Issue by mk30** [rreusser/regl-gpu-lines#3]
i cloned this repo & did `npm install` (was hoping to try out some of the examples). got this error:

```
marina@pumpkin:~/dev/regl-gpu-lines$ npm install
npm ERR! code 1
npm ERR! path /home/marina/dev/regl-gpu-lines/node_modules/gl
npm ERR! command failed
npm ERR! command sh -c prebuild-install || node-gyp rebuild
npm ERR! (node:13786) [DEP0150] DeprecationWarning: Setting process.config is deprecated. In the future the property will be read-only.
npm ERR! (Use `node --trace-deprecation ...` to show where the warning was created)
npm ERR! Package xi was not found in the pkg-config search path.
npm ERR! Perhaps you should add the directory containing `xi.pc'
npm ERR! to the PKG_CONFIG_PATH environment variable
npm ERR! No package 'xi' found
npm ERR! gyp: Call to 'pkg-config --libs-only-L --libs-only-other x11 xi xext' returned exit status 1 while in angle/src/angle.gyp. while loading dependencies of binding.gyp while trying to load binding.gyp
npm ERR! gyp ERR! configure error 
npm ERR! gyp ERR! stack Error: `gyp` failed with exit code: 1
npm ERR! gyp ERR! stack     at ChildProcess.onCpExit (/home/marina/dev/regl-gpu-lines/node_modules/node-gyp/lib/configure.js:351:16)
npm ERR! gyp ERR! stack     at ChildProcess.emit (node:events:365:28)
npm ERR! gyp ERR! stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:290:12)
npm ERR! gyp ERR! System Linux 5.4.0-90-generic
npm ERR! gyp ERR! command "/usr/local/bin/node" "/home/marina/dev/regl-gpu-lines/node_modules/.bin/node-gyp" "rebuild"
npm ERR! gyp ERR! cwd /home/marina/dev/regl-gpu-lines/node_modules/gl
npm ERR! gyp ERR! node -v v16.1.0
npm ERR! gyp ERR! node-gyp -v v7.1.2
npm ERR! gyp ERR! not ok

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/marina/.npm/_logs/2021-11-16T22_39_07_600Z-debug.log
```

my node version is 16.1.0 and my npm version is 7.12.1

any idea what might be going on?

**Comment by rreusser** [comment]
Hmm… it looks like that's an issue with the headless-gl module used for render tests.

Are you able to npm-install `gl` in a blank project? See: https://github.com/stackgl/headless-gl#supported-platforms-and-nodejs-versions

Although it's obnoxious, this shouldn't be necessary to run the examples. If you delete `gl` from package.json, you should be able to run the examples. It's not documented, but you can type `npm start <example-name>` where example-name corresponds to an example in `examples/example-name.js`. It uses budo so should live-reload.

**Comment by rreusser** [comment]
(Also, I'm fighting through one of the last major difficult parts now! Couple hours and inserting breaks should be a little easier to work with. Then I think I'll be looking at version 1.0!🤞 😬 )

**Comment by mk30** [comment]
turns out i can't install `gl` in a blank project either. very strange.

i removed `gl` from the package.json and was able to install properly after that.

and running `npm start multiple` successfully launched the example.

thanks and good luck with the part you're working on now!

**Comment by rreusser** [comment]
Just published v0.0.21! This was one of the last major tricks I wanted to work through: inserting end caps. Now when a position has w = 0, it'll draw a break an insert a cap like this (full disclosure: giving every segment enough vertices for a rounded cap wastes vertices but there's not much choice without geometry shaders; you can always leave `insertCaps: false` and build your endpoint segments manually):

![Screen Shot 2021-11-16 at 20 51 30](https://user-images.githubusercontent.com/572717/142136883-b1dd3a19-e879-4dba-ab60-744ff6d76c0a.png)

I've been using user-specified w = 0 in order to signal a break (now with inserted cap 🎉 ). I think you mentioned that maybe you'd used duplicated points to signal line breaks. Oof, need to think over whether that'd be preferable, but feel free to let me know if anything jumps out as weird, confusing or problematic when you're trying it out!

**Comment by rreusser** [comment]
If it's alright, I'll close this issue since it's not an issue with this project itself, but please feel free to follow up with headless-gl details if you figure it out or with other issues—whether for bugs, features, or just clarification. 🙇
